src/app/services/covid-data-api.service.ts

```diff
@@ -52,52 +52,30 @@ export class CovidDataApiService implements CovidDataApiServiceInterface {
         `Method: ${method.name}() not found in Object: ${this.constructor.name}`
       );
     }
-    return this.apiServices.reduce<Observable<T>>((result, service) => {
-      return result !== EMPTY
-        ? result // result has already been found, skip
-        : service[method.name](...args) // call the method on the subservice, with the args
-            .pipe(
-              catchError((err) => {
-                this.handleError(err);
-                return EMPTY;
-              })
-            );
-    }, EMPTY);
+    let errorCount = 0; // keep track of number or errored out requests for later
+    return race<T>( // only need result from one of the apis
+      this.apiServices.map<Observable<T>>((s, i, arr) => 
+        // call the wanted method from each api
+        s[method.name](...args).pipe( 
+          catchError((err) => {
+            errorCount += 1;
```

Comment:
I think errorCount has a race condition here. Say we have api1 and api2 that both throw an error, depending on the order of operations you can have
1. api1 reads (errorCount = 0)
2. api2 reads (0)
3. api1 increments (1)
4. api2 increments (1)
5. api1 saves (1)
6. api2 saves (1)

so the error count is 1 instead of 2. If you change the order that api2 reads after api1 saves, you'll get a 2.
In java you can solve this using atomic integers, not sure how you can handle this in javascript.

Reply:
Would've been actually, had to look up some remedies but looks like its fine in JS. No need to worry about race conditions
https://stackoverflow.com/questions/5346694/how-to-implement-a-lock-in-javascript

Reply:
this'll probably change though if we move away from using race